x-pack/plugins/apm/public/hooks/use_data_view_id.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { useEffect, useState } from 'react';
+import { getDataViewId } from '../../common/data_view_constants';
+import { ApmPluginStartDeps } from '../plugin';
+
+export function useDataViewId() {
+  const [dataViewId, setDataViewId] = useState<string>(
+    getDataViewId('default')
+  );
+  const { spaces } = useKibana<ApmPluginStartDeps>().services;
+
+  useEffect(() => {
+    const fetchSpaceId = async () => {
+      const space = await spaces?.getActiveSpace();
+      const spaceId = space?.id ?? 'default';
```

Comment:
You probably want to use `DEFAULT_SPACE_ID` from `@kbn/spaces-plugin/common` instead of `default`?! also in the state definition above on like 15?